WDE.SmartScriptEditor/Editor/ViewModels/SmartDataEditors/SmartDataGroupsEditorViewModel.cs

```diff
@@ -43,6 +46,19 @@ public class SmartDataGroupsEditorViewModel: BindableBase, IDocument
             DeleteItem = new DelegateCommand<object>(DeleteItemFromSource);
             AddMember = new DelegateCommand<SmartDataGroupsEditorData>(AddItemToGroup);
             EditItem = new DelegateCommand<SmartDataGroupsEditorData>(EditSourceItem);
+            // history setup
+            History = historyManager;
+            historyHandler = new SmartDataGroupsHistory(SourceItems);
+            UndoCommand = new DelegateCommand(History.Undo, () => History.CanUndo);
+            RedoCommand = new DelegateCommand(History.Redo, () => History.CanRedo);
+            History.PropertyChanged += (sender, args) =>
+            {
+                UndoCommand.RaiseCanExecuteChanged();
+                RedoCommand.RaiseCanExecuteChanged();
+                IsModified = !History.IsSaved;
+                RaisePropertyChanged(nameof(IsModified));
```

Comment:
Since IsModified is driven by History now, you shouldn’t modify it in other places now.